app/src/main/java/com/yacgroup/yacguide/TourbookActivity.kt

```diff
@@ -132,17 +118,70 @@ class TourbookActivity : BaseNavigationActivity() {
         }
     }
 
-    private fun _import() {
-        _ioOption = IOOption.eImport
-        if (FilesystemUtils.checkPermission(this, Manifest.permission.READ_EXTERNAL_STORAGE)) {
-            _showFileChooser()
+    @Throws(IOException::class)
+    private fun _readTextFromUri(uri: Uri): String {
+        val stringBuilder = StringBuilder()
+        contentResolver.openInputStream(uri)?.use { inputStream ->
+            BufferedReader(InputStreamReader(inputStream)).use { reader ->
+                var line: String? = reader.readLine()
+                while (line != null) {
+                    stringBuilder.append(line)
+                    line = reader.readLine()
+                }
+            }
+        }
+        return stringBuilder.toString()
+    }
+
+    private fun _writeTextToUri(uri: Uri, text: String) {
+        try {
+            contentResolver.openFileDescriptor(uri, "w")?.use {
+                FileOutputStream(it.fileDescriptor).use {
+                    it.write(text.toByteArray())
+                }
+            }
+        } catch (e: FileNotFoundException) {
+            e.printStackTrace()
+        } catch (e: IOException) {
+            e.printStackTrace()
```

Comment:
Since you don't return a value or mitigate the exception to the calling function, the user will see a success message in that code path. For simplicity I would make the `_writeTextToUri` method a throwing method and catch the `IOExecption` in `_export()` just as the `JSONException`